routes/draftRoutes.js

```diff
@@ -14,109 +10,112 @@ var router = express.Router();
-router.get('/draft/test/:record_id([A-Fa-f0-9]{24})', permissions.checkPermission("tests:submit"), async function(req, res, next)
+// Edit an existing draft test
+router.get('/draft/test/:testId([A-Fa-f0-9]{24})', permissions.checkPermission("tests:submit"), async function(req, res, next)
 {
   try
   {
-    logger.info("Resume editing draft", req.params.record_id);
+    // Retrieve the draft test's DB entry, using its test ID
+    // If there is no DB entry corresponding to this test ID, just move on
+    var test = await Tests.retrieve(req.params.testId);
     
-    // Get the draft test record
-    var testdata = await Tests.retrieve(req.params.record_id);
-    
-    if(!testdata)
+    if(!test)
     {
       next();
     }
     
-    if(testdata.state != "draft")
+    // Inform the user if the provided test ID does not correspond to a draft
+    // This could happen if the test was in fact submitted, but has somehow also still remained in its original draft state in the DB
+    if(test.state != "draft")
     {
-      return res.status(400).send("Data is not a draft");
+      return res.status(400).send("This test (ID: " + req.params.testId + ") is not a draft!");
     }
     
-    logger.info(testdata);
+    // Get the draft test's type form's ID
+    // Throw an error if the form's ID cannot be found within the draft test's DB entry
+    var formId = test.formId;
     
-    if(!testdata.formId)
+    if(!formId)
     {
-      return res.status(400).send("Cannot find test data with form ID:" + testdata.formId);
+      return res.status(400).send("This draft test (ID: " + req.params.testId + ") has no type form ID!");
     }
 
-    // Get the test type form associated with the given form ID
-    var form = await Forms.retrieve('testForms',testdata.formId);
+    // Retrieve the test's type form, using its form ID
+    // Throw an error if there is no DB entry corresponding to this form ID
+    var form = await Forms.retrieve('testForms', formId);
     
     if(!form)
     {
-      return res.status(400).send("This form ID does not correspond to an existing test type form!");
+      return res.status(400).send("There is no DB entry for a test type form with ID: " + formId);
     }
     
-    // Render the single-component test page
-    res.render('test_run_singleComponent.pug', {formId: testdata.formId,
-                                                form: form,
-                                                componentUuid: testdata.componentUuid,
-                                                testdata: testdata});
+    // Render the page for running a test on a single component (i.e. continue to run this test on the same component)
+    res.render('test_run_singleComponent.pug', {testdata: test,
+                                                formId,
+                                                form,
+                                                componentUuid: test.componentUuid});
   }
   catch(err)
   {
-    logger.error(err);
```

Comment:
Sorry, I should have clarified!  Server-side errors write to `stderr`!  It’s browser/client-side console statements that should be removed.  If we need to log into a server to troubleshoot an issue, logging information is helpful.

Reply:
So then I shouldn't get rid of the `logger` statements?  Only the `console` and `debugger` ones?

Reply:
Yeah :)

Reply:
I've added the logger error messages back in where appropriate.  Note that I've done so only in `routes/jobRoutes.js` ... I've actually already removed this specific file (`routes/draftRoutes.js`) as part of the next commit I'm going to make.